### ISSUE TITLE
Need to filter record.params too

### ISSUE THREAD

**Issue by haridsv** [relaxdiego/relaxdiego.github.com#10]
When the `filter()` function is called, the `record.msg` is not yet formatted, so for something like this:

```
logging.root.info("Got value: %s", value)
```

if value has secret, then it will not get filtered. To be complete, you would have to also handle `record.args`, which can either be a `dict` or a `tuple`. I think a simpler solution is to create custom formatters that would apply the filter on the return value of `Formatter.format()`.

**Comment by haridsv** [comment]
I got the below working:

```
class FilterFormatter(object):
    def __init__(self, orig_filter, secrets):
        self.orig_filter = orig_filter
        self.secrets = secrets

    def format(self, record):
        s = self.orig_filter.format(record)
        for secret in self.secrets:
            s = s.replace(secret, "***")
        return s

    def __getattr__(self, attr):
        return getattr(self.orig_filter, attr)
```

Once the logging is configured, you can wrap all formatters from root logger using the below code:

```
for h in logging.root.handlers:
    h.setFormatter(FilterFormatter(h.formatter))
```

**Comment by relaxdiego** [comment]
Thanks for this bug, @haridsv! I agree. Yours is a much cleaner solution. In fact, it's a more appropriate use of the logging framework since filters should really just decide if the LogRecord should go through or not.

Would you mind filing a PR against the article and becoming said article's co-author? You can put your name at the top similar to how @peshay did it for https://www.relaxdiego.com/2016/06/writing-ansible-modules-with-tests.html

**Comment by haridsv** [comment]
Sure, will give it a shot!

**Comment by relaxdiego** [comment]
Fixed by #11
